src/BackupChain.cs

```diff
@@ -98,22 +98,8 @@ private static IEnumerable<BackupMetadata> NextLogBackup(IEnumerable<BackupMetad
 
     private static bool IsValidFilePath(BackupMetadata meta)
     {
-      if(BackupFileTools.IsUrl(meta.PhysicalDeviceName))
-        return true;
-
-      // A quick check before leaning on exceptions
-      if(Path.GetInvalidPathChars().Any(meta.PhysicalDeviceName.Contains))
-        return false;
-
-      try {
-        // This will throw an argument exception if the path is invalid
-        Path.GetFullPath(meta.PhysicalDeviceName);
-        // A relative path won't help us much if the destination is another server. It needs to be rooted.
-        return Path.IsPathRooted(meta.PhysicalDeviceName);
-      }
-      catch(Exception) {
-        return false;
-      }
+      var path = meta.PhysicalDeviceName;
+      return BackupFileTools.IsValidFileUrl(path) || BackupFileTools.IsValidFilePath(path);

```

Comment:
Does `IsValidFilePath` ever return true here when `IsValidFileUrl` returns false?
I thought the `IsUnc` check took care of the filepath check

Reply:
yes, `IsValidFilePath()` now only validates syntax for local files like these:
- `/some/file.ext` 
- `C:\dir\file.ext`

^^ but `IsValidFileUrl()` (which now contains the UNC check) would return false for these 

I'm assuming it's cause this tool could be running on the same box where backups are stored?
Also another usecase for `IsValidFileUrl()` is here to distinguish the type of path:
- https://github.com/factset/AgDatabaseMove/pull/64/files#diff-b26b234eb55bcea6a1c5cd567db2b6f7e7db565e30a58c70ac62cb22ddf722b1R135
- https://github.com/factset/AgDatabaseMove/pull/64/files#diff-b26b234eb55bcea6a1c5cd567db2b6f7e7db565e30a58c70ac62cb22ddf722b1R210

Reply:
I've left the logic the same as what it was before for `IsValidFilePath()` (except for adding a check for file extensions)